src/transformers/models/seamless_m4t/modeling_seamless_m4t.py

```diff
@@ -0,0 +1,3944 @@
+        vocoder_tgt_lang_id = kwargs.pop("vocoder_tgt_lang_id", None)
+
+        kwargs_text = {"decoder_input_ids": kwargs.pop("decoder_input_ids", None)}
+        kwargs_speech = {}
+        for key, value in kwargs.items():
+            if key.startswith("text_"):
+                key = key[len("text_") :]
+                kwargs_text[key] = value
+            elif key.startswith("speech_"):
+                key = key[len("speech_") :]
+                kwargs_speech[key] = value
+            else:
+                # If the key is already in a specific config, then it's been set with a
+                # submodules specific value and we don't override
+                if key not in kwargs_text:
+                    kwargs_text[key] = value
+                if key not in kwargs_speech:
+                    kwargs_speech[key] = value
+
+        kwargs_text["output_hidden_states"] = True
+        kwargs_text["return_dict_in_generate"] = True
+        kwargs_text["output_scores"] = True
+
+        text_generation_output = super().generate(input_ids, **kwargs_text)
+
+        batch_size = len(input_ids)
+        num_return_sequences = len(text_generation_output.sequences) // batch_size
+        sequences = text_generation_output.sequences
+
+        attention_mask = kwargs_speech.get("attention_mask", kwargs_text.get("attention_mask", None))
+
+        encoder_hidden_states = text_generation_output.encoder_hidden_states[-1]
+
+        # compute last hidden state
+        t2u_input_embeds = self.text_decoder(
+            input_ids=sequences,
+            encoder_hidden_states=encoder_hidden_states,
+            encoder_attention_mask=attention_mask,
+            head_mask=kwargs_text.get("decoder_head_mask"),
+            cross_attn_head_mask=kwargs_text.get("cross_attn_head_mask"),
+        ).last_hidden_state
+
+        # take care of num_return_sequences
+        # take most probable hidden states per batch of return_sequences
+        # (batch_size*num_return_sequences, ...) -> (batch_size,...)
+        if num_return_sequences > 1:
+            idx_most_probable_sequences_per_batch = text_generation_output.sequences_scores.view(
+                batch_size, -1
+            ).argmax(-1)
+            idx_most_probable_sequences_per_batch = (
+                idx_most_probable_sequences_per_batch + torch.arange(batch_size) * num_return_sequences
+            )
+            t2u_input_embeds = t2u_input_embeds[idx_most_probable_sequences_per_batch]
+            sequences = sequences[idx_most_probable_sequences_per_batch]
+
+        pad_token_id = self.generation_config.pad_token_id
+
+        # Compute new attention mask
+        seq_lens = (sequences != pad_token_id).int().sum(1)
+        t2u_model_attention_mask = to_attention_mask(t2u_input_embeds, seq_lens)
+        kwargs_speech["attention_mask"] = t2u_model_attention_mask
+
+        # Compute decoder_input_ids if necessary
+        tgt_lang_id = kwargs_speech.pop("tgt_lang_id", None)
```

Comment:
Do we have to pass an id, or can we pass a language string like Whisper? https://github.com/huggingface/transformers/blob/3fb1535b09901db72a41095c007c29bcdf02e3ae/src/transformers/models/whisper/modeling_whisper.py#L1593

This would be more intuitive for the user no?